From 921829ba19bdde53098b3a0c3141550c909c5a34 Mon Sep 17 00:00:00 2001
From: Guan Jing <guanjing6@huawei.com>
Date: Wed, 30 Nov 2022 18:23:25 +0800
Subject: [PATCH 18/19] sched/fair: Fix kabi borken in sched_domain_shared

hulk inclusion
category: bugfix
bugzilla: https://gitee.com/openeuler/kernel/issues/I61E4M
CVE: NA

--------------------------------

The sched_domain_shared structure is only used as pointer, and other
drivers don't use it directly.

Signed-off-by: Guan Jing <guanjing6@huawei.com>
Reviewed-by: zhangjialin <zhangjialin11@huawei.com>
Reviewed-by: Zhang Qiao <zhangqiao22@huawei.com>
Reviewed-by: Chen Hui <judy.chenhui@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Xue Sinian <xuesinian@huawei.com>
---
 include/linux/sched/topology.h | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/include/linux/sched/topology.h b/include/linux/sched/topology.h
index 0bc030657db4..9bb1d067d84b 100644
--- a/include/linux/sched/topology.h
+++ b/include/linux/sched/topology.h
@@ -80,10 +80,12 @@ struct sched_domain_shared {
 	atomic_t	ref;
 	atomic_t	nr_busy_cpus;
 	int		has_idle_cores;
-	int		nr_idle_scan;
 #if defined(CONFIG_SCHED_STEAL) && !defined(__GENKSYMS__)
 	struct sparsemask *cfs_overload_cpus;
 #endif
+#ifndef __GENKSYMS__
+	int		nr_idle_scan
+#endif
 };
 
 struct sched_domain {
-- 
2.23.0

